addons/account/models/account_payment_term.py

```diff
@@ -234,6 +234,8 @@ def _get_last_discount_date(self, date_ref):
 
     def _get_last_discount_date_formatted(self, date_ref):
         self.ensure_one()
+        if not date_ref:
+            return True
```

Comment:
I'm not sure, returning True here is the best solution for this function. For me, if you don't receive any date, you should return None